### PR TITLE
chore: docker build cache

### PR DESCRIPTION
**Pull request by fafucoder**
**What type of PR is this?**
/kind document

**What this PR does / why we need it**:
At present, the basic principle of docker using cache is that if the parent node has a cache, the current dockerfile statement has been built before and can use cache. So, to speed up compilation, some basic operations should be advanced.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,6 +12,8 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+DOCKER_TAG=latest
```
Comment (zryfish) [inline Makefile:15]: we are tagging image based on which branch we are on

File: build/hypersphere/Dockerfile [hunk 2]
```diff
@@ -2,14 +2,15 @@
 # Use of this source code is governed by an Apache license
 # that can be found in the LICENSE file.
 
+FROM alpine:3.9
```
Comment (zryfish) [inline build/hypersphere/Dockerfile:5]: hypersphere is deprecated, please just delete it

File: hack/docker_build.sh [hunk 3]
```diff
@@ -14,11 +14,14 @@ tag_for_branch() {
 # push to kubespheredev with default latest tag
 REPO=${REPO:-kubespheredev}
 TAG=$(tag_for_branch $1)
+RELEASE=${2:-false}
```
Comment (zryfish) [inline hack/docker_build.sh:17]: I don't see why `release` is necessary here, what you really need to change is `tag_for_branch` function, maybe return a default tag when no branch specified

**Comment by zryfish** [comment]
Please sign your commit with `git commit -s `, also need to add more details about this PR, why we need it

File: Makefile (hunk 1 above)
Comment (fafucoder) [reply Makefile:15]: You are tag_for_branch() function need tag pararms, but you did't pass it

File: hack/docker_build.sh (hunk 3 above)
Comment (fafucoder) [reply hack/docker_build.sh:17]: make build only build image 
make release pull image to repo

use release params to distinguish them

File: build/ks-apiserver/Dockerfile [hunk 4]
```diff
@@ -7,12 +7,13 @@ ENV UID=1002
...
-
 RUN apk add --no-cache ca-certificates && \
     addgroup --gid "$GID" "$USER" && \
-    adduser -D -g "" --ingroup "$USER" -u "$UID" "$USER" && \
-    chown -R kubesphere:"$GID" /usr/local/bin/ks-apiserver
+    adduser -D -g "" --ingroup "$USER" -u "$UID" "$USER"
+
+COPY  /bin/cmd/ks-apiserver /usr/local/bin/
+
+RUN chown -R kubesphere:"$GID" /usr/local/bin/ks-apiserver
```
Comment (LinuxSuRen) [inline build/ks-apiserver/Dockerfile:16]: An extra command will increase the layer of the docker image. I'd prefer to keep them together.

File: build/ks-controller-manager/Dockerfile [hunk 5]
```diff
@@ -3,10 +3,10 @@
 # that can be found in the LICENSE file.
 FROM alpine:3.11
 
-COPY  /bin/cmd/controller-manager /usr/local/bin/
-
 RUN apk add --no-cache ca-certificates
 
+COPY  /bin/cmd/controller-manager /usr/local/bin/
```
Comment (LinuxSuRen) [inline build/ks-controller-manager/Dockerfile:8]: Are there any reasons why we should change it? It'd good if you can explain it. Thanks.

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Hi @fanyanming2016 , Thanks for your contribution. I left some comments, please take a look at it.

**Pull request closed without merging**